src/AddressRegistry/StreetName/StreetNameAddress.cs

```diff
@@ -92,6 +92,19 @@ private void RejectBecauseParentWasRejected()
             }
         }
 
+        private void RejectBecauseParentWasRetired()
+        {
+            if (IsRemoved)
+            {
+                return;
+            }
+
+            if (Status == AddressStatus.Proposed)
+            {
+                Apply(new AddressWasRejectedBecauseHouseNumberWasRetired(_streetNamePersistentLocalId, AddressPersistentLocalId));
+            }
+        }
+
```

Comment:
Can you move this under `RetireBecauseParentWasRetired`

Reply:
Done